File: core/schemas/entity.py
from datetime import datetime, date
from typing import Optional
from pydantic import Field
from .base import BaseSchema


class UserProfile(BaseSchema):
    user_id: str
    name: str
    email: Optional[str] = None
    phone: Optional[str] = None
    date_of_birth: Optional[date] = None
    kyc_status: str
    kyc_verified_at: Optional[datetime] = None
    risk_tier: str
    account_age_days: int
    country: str
    created_at: datetime = Field(default_factory=datetime.utcnow)


class MerchantProfile(BaseSchema):
    merchant_id: str
    name: str
    category: str
    mcc_code: str
    risk_tier: str
    country: str
    registration_date: datetime
    is_verified: bool


class DeviceFingerprint(BaseSchema):
    device_id: str
    user_id: str
    device_type: str
    os: str
    browser: Optional[str] = None
    screen_resolution: Optional[str] = None
    language: Optional[str] = None
    timezone: Optional[str] = None
    is_emulator: bool = False
    is_rooted: bool = False
    first_seen: datetime
    last_seen: datetime


class AccountInfo(BaseSchema):
    account_id: str
    user_id: str
    account_type: str
    bank_code: str
    balance: float
    currency: str
    status: str
    opened_at: datetime

File: apps/dashboard/src/types/entity.ts
export interface UserProfile {
  user_id: string;
  name: string;
  email?: string;
  phone?: string;
  date_of_birth?: string;
  kyc_status: "verified" | "pending" | "rejected" | "not_started";
  kyc_verified_at?: string;
  risk_tier: "low" | "medium" | "high";
  account_age_days: number;
  country: string;
  created_at: string;
}

export interface MerchantProfile {
  merchant_id: string;
  name: string;
  category: string;
  mcc_code: string;
  risk_tier: "low" | "medium" | "high";
  country: string;
  registration_date: string;
  is_verified: boolean;
}

export interface DeviceFingerprint {
  device_id: string;
  user_id: string;
  device_type: string;
  os: string;
  browser?: string;
  screen_resolution?: string;
  language?: string;
  timezone?: string;
  is_emulator: boolean;
  is_rooted: boolean;
  first_seen: string;
  last_seen: string;
}
